frontend/appflowy_flutter/packages/appflowy_editor/test/service/shortcut_event/shortcut_event_test.dart

```diff
@@ -219,5 +219,42 @@ void main() async {
         Selection.single(path: [1], startOffset: text.length),
       );
     });
+
+    //TODO(JRS296): How to test Italicized state of textnode
+    testWidgets('Test Single Asterisk for Italics', (tester) async {
+      const text = '*Welcome to Appflowy 😁';
+      final editor = tester.editor
+        ..insertTextNode(text)
+        ..insertTextNode(text);
+      await editor.startTesting();
+      await editor.updateSelection(
+        Selection.single(path: [1], startOffset: text.length),
+      );
+      if (Platform.isWindows || Platform.isLinux || Platform.isMacOS) {
+        await editor.pressLogicKey(
+          LogicalKeyboardKey.digit8,
+          isShiftPressed: true,
+        );
+      }
+
+      expect(
+        editor.documentSelection,
+        Selection.single(path: [1], startOffset: text.length),
+      );
+      await editor.updateSelection(
+        Selection.single(path: [1], startOffset: 0),
+      );
+
+      if (Platform.isWindows || Platform.isMacOS || Platform.isLinux) {
+        await editor.pressLogicKey(
+          LogicalKeyboardKey.digit8,
+          isShiftPressed: true,
+        );
+      }
+      expect(
+        editor.documentSelection,
+        Selection.single(path: [1], startOffset: 0),
+      );
+    });
```

Comment:
@JRS296

Reply:
Ohh okay, I can try this.
I had planned on going about a test via checking the TextNode's length after conversion to italics.
i.e. 

> *Hello World + *  ----> _Hello World_  

a check for change in length from 12 to 11. Would this be feasible as another extended test? @Xazin

Reply:
It's better to test the attributekeys, testing that a space was removed isn't very telling.

Reply:
If the shift + digit8 works then that's fine.

I'll look at it tomorrow to see what the issue is.

Reply:
Actually, we shouldn't use `shift + digit8` to represent an `asterisk`. Because there're so many keyboard types. We should insert the `asterisk` directly to test this function. Let me write a code sample.

Reply:
It still gives the error specified above on changing to asterisk, please do share any insights for the same, thanks.

Reply:
https://github.com/AppFlowy-IO/AppFlowy/pull/2122/files#diff-79e2eb87335de8f6ca78be7903a1b5fdab9a33b420f77244f1faa6481fc0d75c

Reply:
Please rebase the main branch after the above PR has been merged. Then you can use this function to mock typing a character `*`.

```dart
testWidgets('Test Single Asterisk for Italics', (tester) async {
    const text = '*Hello World';
    final editor = tester.editor..insertTextNode(text);
    await editor.startTesting();

    await editor.updateSelection(
      Selection.single(path: [0], startOffset: text.length),
    );

    await editor.pressLogicKey(character: '*');

    final textNode = editor.nodeAtPath([0]) as TextNode;
    final allBold = textNode.allSatisfyBoldInSelection(
      Selection.single(
        path: [0],
        startOffset: 0,
        endOffset: text.length - 1, // delete the first *
      ),
    );
    expect(allBold, true);
  });

  testWidgets(
      'nothing happens if there\'re more than one * precede the current position',
      (tester) async {
    const text = '**Hello World';
    final editor = tester.editor..insertTextNode(text);
    await editor.startTesting();

    await editor.updateSelection(
      Selection.single(path: [0], startOffset: text.length),
    );

    await editor.pressLogicKey(character: '*');

    final textNode = editor.nodeAtPath([0]) as TextNode;
    final allBold = textNode.allSatisfyBoldInSelection(
      Selection.single(
        path: [0],
        startOffset: 0,
        endOffset: text.length + 1, // insert a new *
      ),
    );
    expect(allBold, false);
    expect(textNode.delta.length, text.length + 1);
  });
```